Create a simple example and some documentation

mpourismaiel opened the pull request.

stp-ip commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "themes/syna"]
+	path = themes/syna
+	url = git@github.com:okkur/syna.git
```
Let's use http instead. That way no authentication is required, when updating.

stp-ip commented on README.md:
````diff
@@ -1,2 +1,36 @@
+$ hugo
+```
+
+> Prerequisites: GoLang, Hugo
````
GoLang -> Go

stp-ip commented on README.md:
````diff
@@ -1,2 +1,36 @@
+
+**Production**:
+```
+$ hugo
````
Add explanation, that this will generate the static files for your site in public/.

stp-ip commented on README.md:
````diff
@@ -1,2 +1,36 @@
+├ config.toml # Hugo config file containing general settings and menu configs.
+```
+
+Please read [Syna's documentation](https://github.com/okkur/syna/tree/master/docs) to understand how to add fragments and change page data.
````
Linking to a versioned doc would make it more aligned I think.

stp-ip commented on config.toml:
```diff
@@ -0,0 +1,100 @@
+languageCode = "en-us"
+title = "Okkur - Syna"
+theme = "syna"
+themesDir = "themes"
```
themesDir is unnecessary as we are using the standard location.

stp-ip commented on config.toml:
```diff
@@ -0,0 +1,100 @@
+  [params.analytics.piwik]
+    url = "https://use.okkur.net"
+    id = "13"
+    domain = "syna-demo.okkur.io" # Track subdomains via "*.example.com"
```
Let's remove our specific piwik config aka make the values empty and comment them out additionally.

stp-ip commented on config.toml:
```diff
@@ -0,0 +1,100 @@
+  name = "Another Link"
+  weight = 30
+
+[[menu.footer_social]]
```
Probably best to group the footer_social entries together.

stp-ip commented on content/_index/contact.md:
```diff
@@ -0,0 +1,51 @@
+
+[message]
+  success = "Thank you for awesomely contacting us." # defaults to theme default
+  error = "Message could not be send. Please contact us at mail@example.com instead." # defaults to theme default
```
Not sure, but as we have a default, we could make these messages empty, but keep them for someone to add in.

success = ""
error = ""

stp-ip commented on content/_index/contact.md:
```diff
@@ -0,0 +1,51 @@
+# Only defined fields are shown in contact form
+[fields.name]
+  text = "Your Name *"
+  error = "Please enter your name" # defaults to theme default
```
All errors default to the theme default. So I would probably go with empty "" for now. Makes it easier for us to iterate on them.

stp-ip replied:
text is necessary for now. Maybe we should make that some default to so a field would be enabled by:
```
[fields.name]
```
or
```
[fields.name]
  text = ""
  error = ""
```

mpourismaiel replied:
Text is a required variable in fields. I emptied the error variable and commented it out for good measure.

stp-ip replied:
Yeah currently it's the minimum required as we are checking for it. Any reason, why we couldn't check for the field itself? This would make the config simpler in my eyes.

stp-ip replied:
But nothing needed here. Either we'll fix it within Syna or push it into the editor fragment.

stp-ip commented on content/_index/embed_subscribe.md:
```diff
@@ -0,0 +1,19 @@
+# Mailerlite is one example of a working provider.
+# There are others such as convertkit.
+# Only necessity is for them to use iframes.
+media = '<iframe src="https://landing.mailerlite.com/webforms/landing/m0h9d7" style="border: none; width: 100%; height: 178px;"></iframe>'
```
This is a "production" form. So not sure, that we should keep it within the starter.
Maybe move the actual form to an example form, if mailerlite provides something like that. or we move it to some random one: `/landing/1234`?

stp-ip commented on README.md:
```diff
@@ -1,2 +1,36 @@
+|  └ _index/ # Landing page is in this directory and it's url is changed to **/**.
+|  └ about/ # About page
+├ layouts/ # You can add extra layout files here. A sample custom fragment is available as a sample.
+├ static/ # Your static files are in this directory. Mostly images directory is needed when using Syna.
```
Maybe link to fall-through specific docs too?

mpourismaiel commented:
Updated the branch as you asked. Please review it once more and let me know if there is anything else.

I also added an installation guide since we use git submodules and it's a less known part of git and the commands are not easily found. Also changed the sample-custom-fragment to give the starting point a better look since I removed the embed fragment altogether to lessen the required config and prevent confusion.

stp-ip commented:
LGTM

Added a few issues so we can address them in follow on PRs.
